exp-player/addon/mixins/exp-player-route.js

```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+    _experiment: null,
+    _session: null,
+
+    currentUser: Ember.inject.service(),
+
+    _getExperiment(params) { // jshint ignore: line
+
+    },
+    _getSession(params, experiment) { // jshint ignore: line
+
+    },
+    model(params) {
+        // While a little gross, this ensures all the criteria for participation
+        // are met before the route resolves. This has the benefit that the route's
+        // loading state is active until all of this is complete.
+        return this._getExperiment(params).then((experiment) => {
+            return this._getSession(params, experiment).then((session) => {
+
+                this.set('_experiment', experiment);
+                this.set('_session', session);
+
+                return experiment.getCurrentVersion().then(versionId => {
+                    session.set('experimentVersion', versionId);
+                    return session.save().then(() => {
+                        return this.get('currentUser').getCurrentUser((account) => {
+                            return account.pastSessionsFor(experiment);
```

Comment:
If the same account has multiple profiles, it seems that we would want to restrict the list of sessions by profile ID, not just account. 

I can help with sample queries for this purpose if needed...